### DISCUSSION TITLE
How to manually forward messages to subscribers?

### DISCUSSION THREAD

**Discussion by antonlovric** [moscajs/aedes#807] [Q&A]
Hi! I created my MQTT broker in Express JS. My approach was to start a broker, listen to messages published by clients, process them and then forward them to the clients.
Issue is, because of my approach clients receive two messages. One is the message I published manually, one is (at least how I understand it) forwarded by Aedes from publisher to all other subscribers. Is there a way to stop Aedes from automatically forwarding the message so I can send them out manually?

This is my current setup, and basically the way I understood this library when starting out with this project. Is there something I misunderstood?
```ts
broker.on("publish", (packet, client) => {
if (!client) return

// ...process the message

// publish the message manually
broker.publish(packet.topic,{
// settings
})

})
```

**Comment by robertsLando** [comment]
> My approach was to start a broker, listen to messages published by clients, process them and then forward them to the clients.

This is not really how a broker should work, I mean you should allow broker itself to forward messages as they arrive.

BTW you may be looking for: https://github.com/moscajs/aedes/blob/main/docs/Aedes.md#handler-authorizeforward-client-packet.

**Reply by antonlovric** [reply]
So the broker is used just for forwarding messages? There shouldn't be any additional operations done on the broker side?

**Reply by robertsLando** [reply]
Yeah but what I mean is that usually the main logic is publish/subscribe. A client publish to a topic and the message is forwarded to who subscribed to it. Then you can do the auth logic and what else you want based on the messages received by the broker, maybe I just don't understand what you are tring to archieve.
